src/ln/channelmanager.rs

```diff
@@ -1574,21 +1574,31 @@ impl ChannelManager {
 	/// generating message events for the net layer to claim the payment, if possible. Thus, you
 	/// should probably kick the net layer to go send messages if this returns true!
 	///
+	/// Require to specify expected amount so that we can claim only payment for the correct amount,
```

Comment:
Should probably say something about "why", otherwise I would totally just specify all amounts in case they have free money :).

eg "You must specify the expected amount for this HTLC, and we will only claim HTLCs available within a few percent of the expected amount. This is critical for several reasons - a) it avoids providing senders with 'proof-of-payment' (in the form of the payment_preimage without having provided the full value and b) it avoids certain privacy-breaking recipient-probing attacks which may review payment activity to motivated attackers!".

Reply:
Note that, iirc, but I'm not 100% sure, the spec says we have to allow payents that are slightly greater than our expected value, which probably needs to be updated in the PaymentReceived event docs as well.

Reply:
The spec : 

> if the amount paid is more than twice the amount expected:
    SHOULD fail the HTLC.
    SHOULD return an incorrect_or_unknown_payment_details error.
        Note: this allows the origin node to reduce information leakage by altering the amount while not allowing for accidental gross overpayment.

So updated the check with `received_amount * 2 > expected_amount ; fail htlc`. 
Also took your comment, and PaymentReceived event docs is already referring to the spec on this.